### PR TITLE
proc: implement cleanup/removal  (#7063)

### PR DESCRIPTION
**Pull request by karajan1001**
fix: #7063
Add `cleanup` and `removal` function to ProcessManager

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/proc/manager.py
Comment (pmrowla) [inline dvc/proc/manager.py:108]:
```suggestion
        for process_info in self.processes():
            if process_info.returncode is not None:
                remove(os.path.join(self.wdir, name))
```

No reason to duplicate the iterator exception handling when we can just use `self.processes()`

Comment (karajan1001) [reply dvc/proc/manager.py:108]: `cleanup` has no name here?

**Comment by pmrowla** [comment]
Can we mark this WIP until the other PR is merged (this is blocked until `kill()` is actually implemented)

File: dvc/proc/manager.py
Comment (karajan1001) [reply dvc/proc/manager.py:108]: Currently, the self.processes() don't provide name info. We have several choices:
1. add `name` to ProcessInfo. 
2. return a (name, ProcessInfo) in `processes()` .
3. return `name`/`key` list in `__iter__(self)`.

For myself, I'd more prefer `3`.

Comment (pmrowla) [reply dvc/proc/manager.py:108]: 3 is probably the best option here (see also https://github.com/iterative/dvc/pull/7048/files#r756744379)

Comment (pmrowla) [reply dvc/proc/manager.py:108]: > `cleanup` is at the manager level, and should remove all stale processes. To remove individual named processes, the caller should just use `remove()`

Comment (pmrowla) [reply dvc/proc/manager.py:108]: One thing I was thinking is that eventually `cleanup` may need a `force` parameter, to remove all processes whether or not they are still running, but that's not a requirement for now.

**Comment by pmrowla** [comment]
@karajan1001 looks like the signal handling tests are flaky
```
        if process_info.returncode is None:
            try:
>               os.kill(process_info.pid, sig)
E               PermissionError: [Errno 1] Operation not permitted

/home/runner/work/dvc/dvc/dvc/proc/manager.py:104: PermissionError
```

Since the tests assign dummy integer PIDs for our running and terminated processes, you need mock out the actual `os.kill` calls. In this case, the test for "process doesn't exist" fails because the dummy PID is actually a valid process on the test machine

**Review by pmrowla — APPROVED** [review]

**Pull request merged**
